apis/meta/conditions.go

```diff
@@ -67,3 +67,28 @@ func FilterOutCondition(conditions []metav1.Condition, conditionType string) []m
 	}
 	return newConditions
 }
+
+// ObjectWithStatusConditions is an interface that describes kubernetes resource
+// type structs with Status Conditions
+type ObjectWithStatusConditions interface {
+	metav1.Object
+	GetStatusConditions() *[]metav1.Condition
+}
+
+// SetResourceCondition sets the given condition with the given status,
+// reason and message on a resource.
+func SetResourceCondition(obj ObjectWithStatusConditions, condition string, status metav1.ConditionStatus, reason, message string) {
+	conditions := obj.GetStatusConditions()
+	gen := obj.GetGeneration()
+
+	newCondition := metav1.Condition{
+		Type:               condition,
+		Status:             status,
+		LastTransitionTime: metav1.Now(),
+		Reason:             reason,
+		Message:            message,
+		ObservedGeneration: gen,
```

Comment:
I think we should do that only if we do actually have logic that would depend on proper handling of the `ObservedGeneration` attribute for conditions, we don't AFAIK. I'm actually wondering if that's gonna be backported to v1.19, as right now it renders the current implementation around ObservedGeneration in conditions pretty much useless. Do you think it's worth it considering all that?

Reply:
The only issue I think is that the observed generation will be set correctly when the condition is first added, but then after that the generation will not be updated, so it will have an incorrect value. So I think either we should add a workaround, or just not set the generation yet at all in this method, until it is working correctly.

Reply:
Yes, I realized that's what's going on right now with the missing value updating line. I would actually go with the latter since we don't yet actively use the `ObservedGeneration` attribute for conditions anywhere, so just ignoring that, for now, should probably do.

Reply:
Removed it in a separate commit that can just be reverted at a later time when the bugfix hits a stable version that we can upgrade to.